src/Development/IDE/Core/Rules.hs

```diff
@@ -212,15 +212,29 @@ getParsedModuleRule = defineEarlyCutoff $ \GetParsedModule file -> do
                 haddockParse = do
                     (_, (!diagsHaddock, _)) <-
                         getParsedModuleDefinition hscHaddock opt comp_pkgs file contents
-                    return diagsHaddock
+                    -- Haddock diagnostics are confusing because they don't say Haddock, and are errors.
+                    -- Fix both those properties.
+                    return $ map (\(a,b,c) -> (a,b,fixHaddockWarning c)) diagsHaddock
 
             ((fingerPrint, (diags, res)), diagsHaddock) <-
                 -- parse twice, with and without Haddocks, concurrently
                 -- we cannot ignore Haddock parse errors because files of
                 -- non-interest are always parsed with Haddocks
                 liftIO $ concurrently mainParse haddockParse
 
-            return (fingerPrint, (mergeDiagnostics diags diagsHaddock, res))
+            -- if you have a Haddock warning and a real warning at the same exact location, throw away the Haddock one
```

Comment:
do we need this special logic? I would think that a comment can only have Haddock warnings

Reply:
We run the parser twice. If you have `(` as your source file, it will be a parse error with both Haddock and without, so you'll get the same error twice. This dedupe ensures you only get the warning once.

Reply:
so the comment is a bit misleading, it's only non Haddock warnings that need deduping, right?

Reply:
Comment updated. Is that clearer?